src/facade/dragonfly_connection.h

```diff
@@ -239,6 +240,9 @@ class Connection : public util::Connection {
 
   void SendAsync(MessageHandle msg);
 
+  // Updates memory stats and pooling, must be called for all used messages
+  void RecycleMessage(MessageHandle msg);
```

Comment:
Nit: I never liked this name tbh, how about `UsedMessageAccounting()` or something like that?

Reply:
I doesn't sound like an action and somewhat confusing, I'd stick with the current